src/fungible_token.ts

```diff
@@ -0,0 +1,131 @@
+      TokenFaucet.deploy(signer, {
+        initialFields: {
+          name: Buffer.from(name, 'utf8').toString('hex'),
+          symbol: Buffer.from(symbol, 'utf8').toString('hex'),
+          decimals,
+          supply: MaxU256,
+          balance: MaxU256
+        },
+        issueTokenAmount: MaxU256
+      })
+    )
+
+    await Withdraw.execute(signer, {
+      initialFields: {
+        token: deployResult.contractInstance.contractId,
+        amount: supply
+      },
+      attoAlphAmount: DUST_AMOUNT
+    })
+
+    return deployResult.contractInstance.address
+  }
+
+  static async load(network: Network): Promise<FungibleToken> {
+    return new FungibleToken(network)
+  }
+
+  async mintTx(signer: SignerProvider, value: bigint, tokenAddress: string) {
+    const tokenFaucet = TokenFaucet.at(tokenAddress)
+    const bytecode = Withdraw.script.buildByteCodeToDeploy({
+      token: tokenFaucet.contractId,
+      amount: value
+    })
+
+    const { address, publicKey } = await signer.getSelectedAccount()
+
+    const unsignedTxBuild = await this.builder.buildExecuteScriptTx(
+      { signerAddress: address, bytecode, attoAlphAmount: DUST_AMOUNT },
+      publicKey
+    )
+    return unsignedTxBuild
+  }
+
+  async mint(signer: SignerProvider, value: bigint, tokenAddress: string) {
+    const buildTxResult = await this.mintTx(signer, value, tokenAddress)
+    const signerAddress = (await signer.getSelectedAccount()).address
+    return await signer.signAndSubmitUnsignedTx({
+      signerAddress: signerAddress,
+      unsignedTx: buildTxResult.unsignedTx
+    })
+  }
+
+  async getAllBalances(tokens: string[], owner: string): Promise<Map<string, bigint>> {
+    const balances = await Promise.all(tokens.map(token => this.balanceOf(owner, token)))
+
+    return new Map(tokens.map((token, i) => [token, balances[i]]))
+  }
+
+  async balanceOf(owner: string, tokenAddress: string): Promise<bigint> {
+    const tokenId = this.getContractId(tokenAddress)
+    return balanceOf(tokenId, owner)
+  }
+
+  async tokenMetadataMulti(tokenAddresses: Array<string>): Promise<Map<string, TokenMetadata>> {
+    const metadata = await Promise.all(
+      tokenAddresses.map((tokenAddress, _) => this.tokenMetadata(tokenAddress))
+    )
+
+    return new Map(tokenAddresses.map((tokenAddress, i) => [tokenAddress, metadata[i]]))
+  }
+
+  async tokenMetadata(tokenAddress: string): Promise<TokenMetadata> {
+    const token = TokenFaucet.at(tokenAddress)
+    const result = await token.multicall({ getSymbol: {}, getName: {}, getDecimals: {} })
+    return {
+      name: hexToString(result.getName.returns),
+      symbol: hexToString(result.getSymbol.returns),
+      decimals: result.getDecimals.returns
+    }
+  }
+
+  async tokenName(tokenAddress: string): Promise<string> {
+    const token = TokenFaucet.at(tokenAddress)
+    return hexToString((await token.view.getName()).returns)
+  }
+
+  async tokenSymbol(tokenAddress: string): Promise<string> {
```

Comment:
should be named `getTokenSymbol`. Same should be in other places in `FungibleToken` class